lib/registration.pm

```diff
@@ -206,26 +228,32 @@ sub register_addons_cmd {
     my ($addonlist) = @_;
     $addonlist //= get_var('SCC_ADDONS');
     my @addons = grep { defined $_ && $_ } split(/,/, $addonlist);
+    if (check_var('DESKTOP', 'gnome') && is_sle('15+')) {
+        my $desk = "sle-module-desktop-applications";
+        record_info($desk, "Register $desk");
+        add_suseconnect_product($desk, undef, undef, undef, 300, 1);
+    }
     foreach my $addon (@addons) {
         my $name = get_addon_fullname($addon);
-        if ($name =~ /adv|containers|legacy|toolchain|web-scripting|public-cloud/) {
-            my @ver = split(/\./, scc_version());
-            add_suseconnect_product($name, $ver[0]);
-        }
-        elsif ($name =~ /live/) {
-            add_suseconnect_product($name, undef, undef, "-r " . get_var('SCC_REGCODE_LIVE'));
-        }
-        elsif ($name =~ /we/) {
-            add_suseconnect_product($name, undef, undef, "-r " . get_var('SCC_REGCODE_WE'));
-        }
-        elsif ($name =~ /LTSS/) {
-            add_suseconnect_product($name, undef, undef, "-r " . get_var('SCC_REGCODE_LTSS'));
-        }
-        elsif ($name =~ /sdk/) {
-            add_suseconnect_product($name);
+        if ($name eq '') {
+            next;
         }
         else {
```

Comment:
since everything you do is only in case `$name` has some value you can just skip branch has `next;` inside and just fallback to 
```suggestion
        if(length $name) {
```